front/lib/actions/mcp_metadata.ts

```diff
@@ -360,18 +360,62 @@ export async function fetchRemoteServerMetaDataByURL(
   });
 
   if (r.isErr()) {
+    return r;
+  }
+  try {
+    return await fetchRemoteServerMetaData(auth, r.value);
```

Comment:
Can we move the try catch inside of that function since we own it?

Part of coding rules to have try catch right around libraries that can throw and turn actionable errors into typed results as soon as possible.